### PR TITLE
Danny R2

### PR DESCRIPTION
**Pull request by dyosorio**
Finished R2 Homework

### CODE REVIEW

**Comment by dyosorio** [comment]
Deployed to Netlify

Demo Exercises: https://vibrant-swartz-f243b0.netlify.app/
Demo Project: https://condescending-curie-742886.netlify.app/

File: react-project/my-app/src/components/.env [hunk 1]
```diff
@@ -0,0 +1,2 @@
+
+REACT_APP_OPENWEATHERMAP_API_KEY=a54b947f262e2c5b975e64258dc64f90
```
Comment (karamad10) [inline react-project/my-app/src/components/.env:2]: env files should not be pushed, add to .gitignore

File: react-project/my-app/src/components/city.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    
+    return (
+        <div>
+            <h1 className='city-name'>{weatherData.name}, {weatherData.sys.country}</h1>
+            <h2>{weatherData.weather[0].main}</h2>
+            <p>{weatherData.weather[0].description}</p>
+            <div className='number-data'>
+                <p className='max-temp'>max temp: {Math.round(weatherData.main.temp_max)}</p>
+                <p className='min-temp'>min temp: {Math.round(weatherData.main.temp_min)}</p>
+                <p>location: {weatherData.coord.lat}, {weatherData.coord.lon}</p>
```
Comment (karamad10) [inline react-project/my-app/src/components/city.js:15]: you're already destructuring from weather data, why not use them?

File: react-project/my-app/src/components/search.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
+import React, { useState } from 'react'
+import City from './city';
+
+const Search = () => {
+    const [query, setQuery] = useState('');
+    const [weatherData, setWeatherData] = useState('');
```
Comment (karamad10) [inline react-project/my-app/src/components/search.js:6]: default state to an object shouldn't be a string, use an empty object

File: react-project/my-app/src/components/search.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+            setLoading(true);
+            fetch(url)
+            .then(res => res.json())
+            .then(data => {
+                if( data.cod === '404'){
+                    setNoResults(true);
+                    setLoading(false);
+                    setWeatherData('')
+                } else {
+                    console.log(data);
```
Comment (karamad10) [inline react-project/my-app/src/components/search.js:25]: console logs are for debugging, make sure you clean all of them when you push your code

File: react-project/my-app/src/components/search.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    const [query, setQuery] = useState('');
+    const [weatherData, setWeatherData] = useState('');
+    const [isLoading, setLoading] = useState(false); 
+    const [hasError, setError] = useState(false);
+    const [noResults, setNoResults] = useState(false);
+
+    const API_KEY = 'a54b947f262e2c5b975e64258dc64f90';
+    const url = `https://api.openweathermap.org/data/2.5/weather?q=${query}&APPID=${API_KEY}&units=metric`
+
+    const fetchWeather = async(e) => {
```
Comment (karamad10) [inline react-project/my-app/src/components/search.js:14]: it's an async function but you're. not using `await` on your data fetch

File: react-project/my-app/src/components/search.js [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+        if(query !== ''){
```
Comment (karamad10) [inline react-project/my-app/src/components/search.js:15]: this is not necessary, get the data and then render based on what you have.

File: week2/my-app/src/components/DogGallery.js [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+*/
+
+import React, { useState,useEffect }  from 'react';
+import DogPhoto from './DogPhoto';
+import DogButton from './DogButton';
+const url='https://dog.ceo/api/breeds/image/random';
+
+function DogGallery() {
+    const [dogPhotos, setDogPhotos] = useState([]);
+    const [clickMessage, setClickMessage] = useState(true);
```
Comment (karamad10) [inline week2/my-app/src/components/DogGallery.js:28]: this should not be a state

File: week2/my-app/src/components/DogGallery.js [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+    const [dogPhotos, setDogPhotos] = useState([]);
+    const [clickMessage, setClickMessage] = useState(true); 
+    const [isLoading, setLoading] = useState(false); 
+    const [hasError, setError] = useState(false);
+
+    const getDogPhoto = async () => {
+        setLoading(true);
+        try {
+            const res = await fetch(url);
+            const data = await res.json();
```
Comment (karamad10) [inline week2/my-app/src/components/DogGallery.js:36]: you have an `await` on fetching the data, you don't need to `await` to format it

File: week2/my-app/src/components/Friends.js [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+    const [friend, setFriend] = useState({});
+    const [data, setData] = useState(false);
+    const [isLoading, setLoading] = useState(false); 
+    const [hasError, setError] = useState(false);
+
+  const getFriend = async () => {
+    setLoading(true);
+    try {
+      const res = await fetch(url);
+      const data = await res.json();
```
Comment (karamad10) [inline week2/my-app/src/components/Friends.js:40]: you have an await on fetching the data, you don't need to await to format it

File: week2/my-app/src/components/Friends.js [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+
+import React, { useState, useEffect }  from 'react';
+import FriendProfile from './FriendProfile';
+import Button from './FriendButton';
+
+const url = 'https://www.randomuser.me/api?results=1';
+
+function Friend() {
+    const [friend, setFriend] = useState({});
+    const [data, setData] = useState(false);
```
Comment (karamad10) [inline week2/my-app/src/components/Friends.js:32]: also no need for this to be a state, you can tell if there's data by checking your `friend` state

File: week2/my-app/src/components/RandomJoke.js [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+
+const RandomJoke = () => {
+    const [joke, setJoke] = useState({});
+    const [clickMessage, setClickMessage] = useState(true); 
+    const [isLoading, setLoading] = useState(false); 
+    const [hasError, setError] = useState(false);
+
+    useEffect(() => {
+        const interval = setInterval(() => {
+            console.log('This will run every 20 second!');
```
Comment (karamad10) [inline week2/my-app/src/components/RandomJoke.js:31]: remove all logs when pushing your code

File: week2/my-app/src/components/RandomJoke.js [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+            setJoke(data);
+            setLoading(false);
+            setClickMessage(false);
+        })
+        .catch(error => {
+            setError(true);
+            setLoading(false);
+        })
+          }, 20000);
+          return () => clearInterval(interval);
```
Comment (karamad10) [inline week2/my-app/src/components/RandomJoke.js:46]: 👍

**Comment by karamad10** [comment]
Hey @dyosorio good job on the hw, I like the styling and the extra efforts.

however there are a few issues that I commented on and I wanna mention that this was very hard to review because you have 2 full-fledged react apps, please be wary of what your pushing, all of the week1 homework in there.

File: react-project/my-app/src/components/.env (hunk 1 above)
Comment (dyosorio) [reply react-project/my-app/src/components/.env:2]: I will, thank you.

File: react-project/my-app/src/components/search.js (hunk 3 above)
Comment (dyosorio) [reply react-project/my-app/src/components/search.js:6]: I understand, thank you for clarifying this.

File: react-project/my-app/src/components/city.js (hunk 2 above)
Comment (dyosorio) [reply react-project/my-app/src/components/city.js:15]: I agree, I'll fix it.

**Comment by dyosorio** [comment]
Hi, @karamad10 Thank you for reviewing my homework and for your comments. I apologize for the commits from week 1, I don't know how to fix that. I'm having the same problem on w3, I ran git rebase and now I'm carrying all the commits from week to week, even though they don't belong to the current branch. 

I appreciate your time and valuable feedback.

**Comment by karamad10** [comment]
Hey @dyosorio , i think the problem is that when you open you VSCode you're on the previous branch, for example, this is week2 and when you create a new branch for week3 everything carries out to the other branch so now if you make a week4 next while on week3 everything will continue to carry with you BUT if you check out to master first and then create your new week3 branch then that would be clean : )
in short check out to master and make a new branch from there every week : ) 
hope this helps and if it doesn't work out or you don't understand what is happening let me know and I can explain more : )

**Comment by dyosorio** [comment]
Hi @karamad10 You're exactly right. I was creating my new branches based on the previous ones and I merged them, that is why I was carrying all the assignments from week to week. I already corrected this issue for React W3 and I opened a new pull request with a clean branch https://github.com/HackYourHomework/React/pull/39, it will be easier to check I hope.

Thank you very much for all your support.
